mvcc/watchable_store.go

```diff
@@ -180,6 +180,29 @@ func (s *watchableStore) cancelWatcher(wa *watcher) {
 	s.mu.Unlock()
 }
 
+func (s *watchableStore) Restore(b backend.Backend) error {
+	var err error
+	var needSync bool
+	func() {
```

Comment:
It's because I have to release the `mutex` before calling `syncWatchers` that I use `func() {} ()`, but as you commented, It will be better not calling `syncWatchers`. so, yes, the codes you posted here is better.